tests/Analyze/Gen.hs

```diff
@@ -194,10 +194,10 @@ genCore (BoundedInt size) = Gen.recursive Gen.choice [
        let size' = unaryArithSize op size
        Gen.subtermM (genCore (BoundedInt size')) $
          mkInt . Numerical . IntUnaryArithOp op . extract
-  , Gen.subtermM (genCore (BoundedDecimal size)) $ \x -> do
-    op <- genRoundingLikeOp
-    mkInt $ Numerical $ RoundingLikeOp1 op (extract x)
-  , Gen.subtermM (genCore strSize) $ mkInt . StrLength . extract
+  , do x <- genCore (BoundedDecimal size)
+       op <- genRoundingLikeOp
+       mkInt $ Numerical $ RoundingLikeOp1 op (extract x)
+  , genCore strSize >>= mkInt . StrLength . extract
```

Comment:
This and the following changes are unrelated. They fix the shrinking logic of the term generation. The previous shrinking was too aggressive and could result in ill typed terms.